agent/unit-test/softwarecontaineragent_unittest.cpp

```diff
@@ -0,0 +1,213 @@
+     }
+};
+} //namespace
+
+class SoftwareContainerAgentTest: public ::testing::Test
+{
+public:
+    SoftwareContainerAgent *sca;
+
+    const std::string configString = "[SoftwareContainer]\n"
+#ifdef ENABLE_NETWORKGATEWAY
+                                     "create-bridge = true\n"
+                                     "bridge-device = " + std::string(BRIDGE_DEVICE_TESTING) + "\n"
+                                     "bridge-ip = " + std::string(BRIDGE_IP_TESTING) + "\n"
+                                     "bridge-netmask = " + std::string(BRIDGE_NETMASK_TESTING) + "\n"
+                                     "bridge-netmask-bitlength = " + std::string(BRIDGE_NETMASK_BITLENGTH_TESTING) +"\n"
+                                     "bridge-netaddr = " + std::string(BRIDGE_NETADDR_TESTING) + "\n"
+#endif
+                                     "shutdown-timeout = 1\n"
+                                     "shared-mounts-dir = " + std::string(SHARED_MOUNTS_DIR_TESTING) + "\n"
+                                     "deprecated-lxc-config-path = " + std::string(LXC_CONFIG_PATH_TESTING) + "\n"
+                                     "service-manifest-dir = " + std::string(SERVICE_MANIFEST_DIR_TESTING) + "\n"
+                                     "default-service-manifest-dir = " + std::string(DEFAULT_SERVICE_MANIFEST_DIR_TESTING) + "\n";
+
+    const std::string valid_config = "[{\"enableWriteBuffer\": false}]";
+
+    void SetUp() override
+    {
+        std::unique_ptr<ConfigLoader> loader(new StringConfigLoader(configString));
+        std::unique_ptr<ConfigSource> mainConfig(new MainConfigSource(std::move(loader),
+                                                                      ConfigDefinition::typeMap()));
+
+        std::vector<std::unique_ptr<ConfigSource>> configSources;
+        configSources.push_back(std::move(mainConfig));
+
+        std::shared_ptr<Config> config = std::make_shared<Config>(std::move(configSources),
+                                                                  ConfigDefinition::mandatory(),
+                                                                  ConfigDependencies());
+
+        Glib::RefPtr<Glib::MainContext> mainContext = Glib::MainContext::get_default();
+        std::shared_ptr<SoftwareContainerFactory> factory = std::shared_ptr<SoftwareContainerFactory> (new TestFactory());
+        sca = new SoftwareContainerAgent(mainContext, config, factory);
+    }
+    void TearDown() override
+    {
+        delete sca;
+    }
+};
+
+/*
+ * Test that an agent instance can create, destroy, and create another container.
+ */
+TEST_F(SoftwareContainerAgentTest, SequenceTest) {
+    ASSERT_NO_THROW({
+        auto id = sca->createContainer(valid_config);
+        sca->suspendContainer(id);
+        sca->resumeContainer(id);
+        sca->getContainer(id);
+        auto containers = sca->listContainers();
+        ASSERT_EQ(1, containers.size());
+
+        sca->shutdownContainer(id);
+    });
+}
```

Comment:
Agree. The purpose is to have tests for the sequence, i.e. that some calls on the agent results in the expected call(s) to SoftwareContainer.